Init CHANGELOG.md (#33)
curl/wcurl: CHANGELOG.md

<!--
Copyright (C) Samuel Henrique <samueloph@debian.org>, Sergio Durigan
Junior <sergiodj@debian.org> and many contributors, see the AUTHORS
file.

SPDX-License-Identifier: curl
-->

# Changelog

## [UNRELEASED]

>  * Default to index.html as filename if non can be parsed from the URL.

Reviewer: I think there's some typo here...?  I don't fully understand the sentence.

 * Percent-decode output filenames by default.
 * New option to disable percent-decoding of output filenames: --no-decode-filename.
 * wcurl.1: Fix typo on list of features.
 * README/manpage: point to the curl issue tracker.
 * README:
   - Add a missing dash to the --dry-run command.
   - Add a logo.
   - Add brief section explaining about our testsuite.
   - Remove HTML <a name> anchors.
 * Symlink LICENSE to LICENSES/curl.txt.

## [v2024.07.10]
 * Change versioning to use dots as separators instead of dashes:
     - Previous version: 2024-07-07
     - New version: 2024.07.10
 * Support older curl releases, minimum required version is now 7.46.0:
     - Only set --no-clobber if curl is 7.83 or newer

>      - Only set --parallel if curl is 7.66 or newer

Reviewer: Missing period at the end of this sentence and the previous one.

>  * Set --fail on curl, in order to return errors instead of saving as

Reviewer: ```suggestion
 * Set `--fail` when invoking curl, in order to display possible errors instead of saving them as
```

   output files.
 * Add more tests.
 * Remove the need for GNU coreutils' realpath for tests.

>  * wcurl.1: Update manpage with links to Github and Debian's Salsa.

Reviewer: This format of `file: what changed` isn't used anywhere else, so I think we shouldn't use it here either.

 * Update LICENSE file with new contributors.

## [v2024-07-07]

>  * Drop getotp usage, non-linux environments are supported now.

Reviewer: - `s/getotp/getopt/`

- Use backticks on `getopt`.

- I will personally use "GNU/Linux", but it's OK if you don't want to.  However, I believe we should at least capitalize the name "Linux" here.

>  * Replace "-o/--opts=" parameters with "--curl-options/--curl-options=".

Reviewer: Since this file is explicitly written in Markdown, we should use backticks here and elsewhere.

   This alternative is more descriptive and it does not coincide with any of curl's parameters.
 * Stop auto-resuming downloads and don't overwrite files instead by default.
   Safer alternative as otherwise curl can corrupt a file if the name clashes and the size of the existing one is smaller.

>    One can easily change that behavior with '--curl-options="--continue-at -'.

Reviewer: There's a mix of single and double quotes here (but we should use backticks).

 * New --dry-run option: just print what would be invoked.

>  * Choose https as a default protocol, in case there's none in the URL.

Reviewer: `s/https/HTTPS/`.

 * Disable curl's URL globbing parser so {} and [] characters in URLs are not treated specially.
 * Implement support for '--'.

>  * Implement a -V and --version options.

Reviewer: ```suggestion
 * Implement `-V/--version` options.
```

 * Basic testsuite implemented.
 * Update manpage, README and help output.

## [v2024-07-02]
 * First release to be shipped on Debian.

## [v2024-06-26]

>  * Second release.

Reviewer: Not that it matters much, but would you like to add some information on what changed here?

<!--
Copyright (C) Samuel Henrique <samueloph@debian.org>, Sergio Durigan
Junior <sergiodj@debian.org> and many contributors, see the AUTHORS
file.

SPDX-License-Identifier: curl
-->

# Changelog

## [UNRELEASED]
 * New parameter `-o|-O|--output|output=` which allows the user to choose the output filename.

>  * Default to "index.html" as filename if none can be inferred from the URL.

Reviewer: ```suggestion
 * Default to `index.html` as filename if none can be inferred from the URL.
```

 * Percent-decode output filenames by default.
 * New option to disable percent-decoding of output filenames: `--no-decode-filename`.
 * Fix typo in the list of features of the manpage.
 * README/manpage: Point to the curl issue tracker.
 * README:
   - Add a missing dash to the `--dry-run` command.
   - Add a logo.
   - Add a brief section explaining about our testsuite.
   - Remove HTML `<a name>` anchors.
 * Symlink LICENSE to LICENSES/curl.txt.

## [v2024.07.10]
 * Change versioning to use dots as separators instead of dashes:

>      - Previous version: 2024-07-07
>      - New version: 2024.07.10

Reviewer: ```suggestion
     - Previous version: `2024-07-07`
     - New version: `2024.07.10`
```

 * Support older curl releases, minimum required version is now 7.46.0:
     - Only set `--no-clobber` if curl is 7.83 or newer.
     - Only set `--parallel` if curl is 7.66 or newer.
 * Set `--fail` when invoking curl, in order to display possible errors instead of saving them as
   output files.
 * Add more tests.

>  * Remove the need for GNU coreutils' realpath for tests.

Reviewer: ```suggestion
 * Remove the need for GNU coreutils' `realpath` for tests.
```

 * Update manpage with links to Github and Debian's Salsa.
 * Update LICENSE file with new contributors.

## [v2024-07-07]
 * Drop `getopt` usage, non-GNU/Linux environments are supported now.

>  * Replace `-o/--opts=` parameters with `--curl-options/--curl-options=`.

Reviewer: ```suggestion
 * Replace `-o`/`--opts=` parameters with `--curl-options`/`--curl-options=`.
```

   This alternative is more descriptive and it does not coincide with any of curl's parameters.
 * Stop auto-resuming downloads and don't overwrite files instead by default.
   Safer alternative as otherwise curl can corrupt a file if the name clashes and the size of the existing one is smaller.
   One can easily change that behavior with `--curl-options="--continue-at -"`.
 * New `--dry-run` option: just print what would be invoked.
 * Choose HTTPS as a default protocol, in case there's none in the URL.
 * Disable curl's URL globbing parser so `{}` and `[]` characters in URLs are not treated specially.

>  * Implement support for `--`'.

Reviewer: ```suggestion
 * Implement support for `--`.
```

>  * Implement `-V/--version` options.

Reviewer: ```suggestion
 * Implement `-V`/`--version` options.
```

 * Basic testsuite implemented.
 * Update manpage, README and help output.

## [v2024-07-02]
 * First "public" release, announcing the project.
 * Use `exec` instead of `eval`.
 * Only set `--parallel` if there's more than one URL.
 * Fix manpage typo.
 * Update COPYRIGHT and AUTHORS in manpage.

>  * Rewrite wcurl to remove bash dependency, it's now a POSIX shellscript.

Reviewer: ```suggestion
 * Rewrite wcurl to remove bash dependency, it's now a POSIX shell script.
```

 * Add README.md.
 * Add LICENSE.

## [v2024-06-26]

>  * Simplify`--help` output.

Reviewer: ```suggestion
 * Simplify `--help` output.
```
